### PR TITLE
Better webhook queue handling.

### LINKED ISSUE #2236: Webhooks getting delayed anywhere from 2 - 45+ minutes
## Expected Behavior
Should be getting sent based off of -whfi

## Current Behavior
webhooks sending extremely delayed, if at all. sending the post to webhook outside of RM works immediately. multiple reports on discord of same issue.

issue seemed to have started at some point around July 15th.

## Webhook settings used:
webhook: [http://127.0.0.1:5112]
wh-types:[pokemon] 
webhook-blacklist: [1,2,4,5,7,8,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,27,29,30,32,33,35,37,39,41,43,44,46,48,50,52,54,56,58,60,61,63,66,69,72,74,77,79,86,90,92,96,98,102,104,111,116,118,119,120,129,133,152,153,155,156,158,161,162,163,164,165,166,167,168,170,177,178,183,184,185,187,188,190,191,193,194,195,198,200,202,203,204,205,206,207,209,210,211,215,216,218,220,223,224,226,227,228,231,234]

## Your Environment

* Version used: Latest
* Environment name and version (e.g. Python 2.7): python 2.7
* Operating System and version (desktop or mobile): Windows 7

git status
negative time - alarm

### PR DESCRIPTION
**Pull request by friscoMad**
## Description
While #2248 does improve the situation we still need to do a timed get to properly fix the issue in setups with low number of triggers.

**Seb:** Fixes #2236.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/webhook.py [hunk 1]
```diff
@@ -76,55 +77,57 @@ def wh_updater(args, queue, key_caches):
...
-                    frame_messages.append(frame_message)
-                else:
-                    # Make sure to call key_cache[ident] in all branches so it
-                    # updates the LFU usage count.
-
-                    # If the object has changed in an important way, send new
-                    # data to webhooks.
-                    if __wh_object_changed(whtype, key_cache[ident], message):
+            try:
+                timeout = frame_interval_sec if num_messages > 0 else None
```
Comment (sebastienvercammen) [inline pogom/webhook.py]: `num_messages` never gets updated.

File: pogom/webhook.py [hunk 2]
```diff
@@ -76,55 +77,57 @@ def wh_updater(args, queue, key_caches):
...
+                whtype, message = queue.get(True, timeout)
+            except Empty:
+                pass
```
Comment (sebastienvercammen) [inline pogom/webhook.py]: If the queue hits timeout and `queue.Empty`, it means `num_messages > 0` and we've waited for `frame_interval_sec`, and so the frame should still be sent.

File: pogom/webhook.py (hunk 1 above)
Comment (friscoMad) [reply pogom/webhook.py]: Line 135 after we process the new wh

File: pogom/webhook.py (hunk 2 above)
Comment (friscoMad) [reply pogom/webhook.py]: And it is sent if does skip the else: clause of the try and check the time and send it as we have waited for it.

Comment (sebastienvercammen) [reply pogom/webhook.py]: You're absolutely right, I misread the block in Github's diff view, which doesn't include the sending.

File: pogom/webhook.py (hunk 1 above)
Comment (sebastienvercammen) [reply pogom/webhook.py]: Same problem as my other comment, it was excluded from the diff view. I think it's telling me to head to bed.

**Comment by friscoMad** [comment]
I did test the previous version that did not use num_messages and used len
directly but not this one I will do some tests tomorrow as I am not sure it
is updated properly when a message is sent.

**Pull request merged**